tools/online_editor/src/properties_widget.ts

```diff
@@ -66,10 +66,18 @@ export class PropertiesWidget extends Widget {
         return this.#language_client;
     }
 
-    position_changed(uri: LspURI, _version: number, position: LspPosition) {
-        query_properties(this.language_client, uri, position, (r) => {
-            this.#propertiesView.set_properties(r);
-        });
+    position_changed(uri: LspURI, version: number, position: LspPosition) {
+        query_properties(this.language_client, uri, position)
+            .then((r: PropertyQuery) => {
+                this.#propertiesView.set_properties(r);
+            })
+            .catch(() => {
+                // Document has not loaded yet!
```

Comment:
This is really unfortunate. Isn't there a way to have a promise that resolves when the document is loaded, and wait on that instead? Like `ensure_document_loaded_promise.then(() => { query_properties... });` or with an `await`? Like we do with `ensure_lsp_running`.

Reply:
Upside is that perhaps there is a way of doing this without a one second timer, but at the very least the 1 second is coded in one place instead of in many.

Reply:
> I am not aware of the LSP sending that information.

It's our LSP, can't we wait for a (custom) message that we send perhaps?